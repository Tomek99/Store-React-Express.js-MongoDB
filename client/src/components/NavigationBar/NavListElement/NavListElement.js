import React from "react";
import { NavLink, Link } from "react-router-dom";
import PropTypes from "prop-types";
import styles from "./NavListElement.module.scss";

function NavListElement({ isLink, name, path }) {
  return (
    <li className={styles.NavListElement}>
      {isLink ? (
        <Link to={path}>{name}</Link>
      ) : (
        <NavLink
          to={path}
          className={({ isActive, isPending }) =>
            isPending
              ? "pending"
              : isActive
              ? styles.activeNavLink
              : styles.navLink
          }
        >
          {name}
        </NavLink>
      )}
    </li>
  );
}

NavListElement.propTypes = {
  isLink: PropTypes.bool,
  name: PropTypes.string,
  path: PropTypes.string,

  style: PropTypes.string,
};
export default NavListElement;
