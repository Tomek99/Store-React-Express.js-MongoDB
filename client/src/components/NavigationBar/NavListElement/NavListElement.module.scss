.NavListElement {
    .navLink {
        color: #fff;
        transition: 0.3s linear;
        position: relative;

        &::after {
            display: block;
            content: "";
            width: 0;
            height: 2px;
            background: var(--main-color);
            position: absolute;
            bottom: -3px;
            left: 0;
            transition: 0.3s linear;
        }

        &:hover::after {
            width: 100%;
        }

        &:hover {
            color: var(--main-color);
        }
    }

    .activeNavLink {
        color: var(--main-color);
        &::after {
            display: block;
            content: "";
            width: 100%;
            height: 2px;
            background: var(--main-color);
            position: absolute;
            bottom: -3px;
            left: 0;
            transition: 0.25s linear;
        }
    }
}
