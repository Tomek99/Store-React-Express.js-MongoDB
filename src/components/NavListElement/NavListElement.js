import React from "react";
import { NavLink, Link } from "react-router-dom";

function NavListElement(props) {
  const { isLink, name, path, activeStyle, style } = props;
  return (
    <li className={style} key={name}>
      {isLink ? (
        <Link to={path}>{name}</Link>
      ) : (
        <NavLink
          to={path}
          className={({ isActive }) => (isActive ? activeStyle : undefined)}
        >
          {name}
        </NavLink>
      )}
    </li>
  );
}

export default NavListElement;
