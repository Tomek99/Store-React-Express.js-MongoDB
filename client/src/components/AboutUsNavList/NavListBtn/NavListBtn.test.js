import { render, screen } from "@testing-library/react";
import NavListElement from "./NavListBtn";

test("should render NavListBtn", async () => {
  const item = { name: "About Us", number: 0 };
  const { container } = render(<NavListElement item={item} tabNumber={0} />);

  expect(container).toMatchSnapshot();
});
