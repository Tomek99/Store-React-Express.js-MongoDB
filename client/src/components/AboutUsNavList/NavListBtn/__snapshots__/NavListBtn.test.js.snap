// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavListBtn 1`] = `
<div>
  <li>
    <button
      class="activeStyle"
    >
      About Us
    </button>
  </li>
</div>
`;
